Docs/Architecture/CommandBased.md

```diff
@@ -2,26 +2,47 @@
+Therefore, it might not be stored in a Subsystem.
+
+On the other hand, a roller that is driven by a motor can only go at one speed at a time.
+Therefore, we would wrap it in a Subsystem so that only one Command can use it at once.
+
+A Trigger is something which can start a Command.
+The classic form of this is a button on the driver's controller.
+Another common type is one which checks for when the robot enables.
+One non-obvious Trigger we used in 2024 was one which checked when we had detected a game piece in the robot, which we used to flash our LEDs and vibrate the driver controller.
+Triggers can be made of any function that returns a boolean which makes them very powerful.
+Some large Commands are better represented by several Commands and some Triggers!
+
 ### Resources
 
+- [WPILib intro to functional programming](https://docs.wpilib.org/en/stable/docs/software/basic-programming/functions-as-data.html).
+  Read through this article on lambda expressions and functional programming if you haven't already.
 - [WPILib docs](https://docs.wpilib.org/en/stable/docs/software/commandbased/index.html).
-  Read through these docs until you finish "The Command Scheduler"
+  Read through these docs until you finish "Organizing Command-Based Robot Projects"
   OR watch [this video](https://drive.google.com/file/d/1ykFDfXVYk27aHlXYKTAqtj1U2T80Szdj/view?usp=drive_link).
-  Presentation notes for the video are [here](CommandBasedPresentationNotes.md)
-  The important segment to remember is:
+  Presentation notes for the video are [here](CommandBasedPresentationNotes.md).
+  If you watch the video, it is recommended to also read the [Subsystems](https://docs.wpilib.org/en/stable/docs/software/commandbased/subsystems.html), [Binding Commands to Triggers](https://docs.wpilib.org/en/stable/docs/software/commandbased/binding-commands-to-triggers.html), and [Organizing Command-Based Robot Projects](https://docs.wpilib.org/en/stable/docs/software/commandbased/organizing-command-based.html#) for addition details on using Command-Based.
+  
+  The important segment of all of this to remember is:
   > Commands represent actions the robot can take. Commands run when scheduled, until they are interrupted or their end condition is met. Commands are very recursively composable: commands can be composed to accomplish more-complicated tasks. See Commands for more info.
   >
   > Subsystems represent independently-controlled collections of robot hardware (such as motor controllers, sensors, pneumatic actuators, etc.) that operate together. Subsystems back the resource-management system of command-based: only one command can use a given subsystem at the same time. Subsystems allow users to “hide” the internal complexity of their actual hardware from the rest of their code - this both simplifies the rest of the robot code, and allows changes to the internal details of a subsystem’s hardware without also changing the rest of the robot code.
```

Comment:
not sure if you want to have this as a block quote, do you also want to add a summary for triggers?

Reply:
these are taken directly from the wpilib docs, hence the block quote. not sure if it still needs to be there since its decently redundant w the intro